### PR TITLE
fix issue with shim use of config.Duration

### PR DESCRIPTION
**Pull request by ssoroka**
The execd shim loads config in a slightly different way. As a result, the config.Duration wasn't being handled properly. 

This PR fixes it, and resolves issues mentioned in https://github.com/influxdata/telegraf/issues/7911

### CODE REVIEW

File: plugins/inputs/vsphere/endpoint.go [hunk 1]
```diff
@@ -887,7 +887,7 @@ func (e *Endpoint) chunkify(ctx context.Context, res *resourceKind, now time.Tim
 			// Determine time of last successful collection
 			metricName := e.getMetricNameForId(metric.CounterId)
 			if metricName == "" {
-				e.log.Info("Unable to find metric name for id %d. Skipping!", metric.CounterId)
+				e.log.Infof("Unable to find metric name for id %d. Skipping!", metric.CounterId)
```
Comment (reimda) [inline plugins/inputs/vsphere/endpoint.go:890]: This is a good change but looks unrelated.  Was it supposed to be in this pr?

File: plugins/processors/reverse_dns/reversedns_test.go [hunk 2]
```diff
@@ -54,3 +54,24 @@ func TestLoadingConfig(t *testing.T) {
 
 	require.Len(t, c.Processors, 1)
 }
+
```
Comment (reimda) [inline plugins/processors/reverse_dns/reversedns_test.go:57]: This test belongs in config/config_test.go instead of here.  TestConfig_LoadSpecialTypes is similar.

File: plugins/inputs/vsphere/endpoint.go (hunk 1 above)
Comment (ssoroka) [reply plugins/inputs/vsphere/endpoint.go:890]: Unrelated; just don't want to make a new pr for one line. :D

File: plugins/processors/reverse_dns/reversedns_test.go (hunk 2 above)
Comment (ssoroka) [reply plugins/processors/reverse_dns/reversedns_test.go:57]: you can't have config require reverse_dns due to cyclical dependencies. Since it mentions both, it has to go here.

Comment (ssoroka) [reply plugins/processors/reverse_dns/reversedns_test.go:57]: .. though, a config_test package might work.

Comment (ssoroka) [reply plugins/processors/reverse_dns/reversedns_test.go:57]: Looking at this with fresh eyes, the test above it is almost identical in setup. I'm not so sure that it belongs in config. The functionality in config is covered by another test (in config_test above), and this was just an extra sanity test I added to make sure the plugin had what it needed, and wasn't falling back on default values. In the end this was not needed (because it's not a regression test per se, and didn't highlight the bug I was fixing), but I figured it didn't hurt to add the test to the test suite. I'd recommend leaving it as is.

Comment (reimda) [reply plugins/processors/reverse_dns/reversedns_test.go:57]: I understand it's convenient to put it here because of dependencies, but it doesn't belong here because it's testing config parsing, not reverse dns functionality.  I'll leave it to you to move if you want.
